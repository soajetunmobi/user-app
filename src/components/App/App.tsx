import React, { FunctionComponent } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { AppLayout } from '../AppLayout'
import Routes from 'routing/Routes'

export const App: FunctionComponent = () => {
  return (
    <BrowserRouter>
      <AppLayout>
        <Routes />
      </AppLayout>
    </BrowserRouter>
  )
}
