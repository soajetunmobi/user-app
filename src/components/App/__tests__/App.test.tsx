import React from 'react'
import renderer from 'react-test-renderer'
import { App } from '../App'

test('it renders correctly', () => {
  const root = renderer.create(<App />).toJSON()
  expect(root).toMatchSnapshot()
})
