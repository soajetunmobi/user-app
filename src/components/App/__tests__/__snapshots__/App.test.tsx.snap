// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="logo"
    >
      <span
        className="cursor--pointer"
        data-testid="logo-link"
        onClick={[Function]}
      >
        User App
      </span>
    </div>
    <div>
      <span
        className="cursor--pointer"
        data-testid="login-link"
        onClick={[Function]}
      >
        Login |
         
      </span>
      <span
        className="cursor--pointer"
        onClick={[Function]}
      >
        Profile
      </span>
    </div>
  </div>,
  <main
    className="content"
  >
    <div>
      <div
        className="text-align--center"
      >
        <h4
          data-testid="login-welcome"
        >
          Welcome, please login to access your profile
        </h4>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          data-testid="login-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
      </div>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <span>
      User App
    </span>
    <br />
    2020
  </footer>,
]
`;
